src/robot/libdocpkg/specbuilder.py

```diff
@@ -75,31 +74,17 @@ def _create_arguments(self, elem):
             ArgInfo.VAR_NAMED: lambda value: setattr(spec, 'var_named', value),
         }
         for arg in elem.findall('arguments/arg'):
-            name = self._get_name(arg)
-            if name:
-                setters[arg.get('kind')](name)
-                spec.defaults.update(self._get_default(arg, name))
-                spec.type = self._get_updated_types(arg, name, spec)
+            name_elem = arg.find('name')
+            if name_elem is None:
+                continue
+            name = name_elem.text
+            setters[arg.get('kind')](name)
+            default_elem = arg.find('default')
+            if default_elem is not None:
+                spec.defaults[name] = default_elem.text or ''
+            type_elem = arg.find('type')
+            if type_elem is not None:
+                if not spec.types:
+                    spec.types = {}
+                spec.types[name] = type_elem.text
```

Comment:
Great that the bug with setting type info is fixed. It would have been good to add a test validating the fix as well but I assume you are going to add more round-trip tests in the future.